react/react1/week2/todo-list/src/TodoItem.jsx

```diff
@@ -0,0 +1,28 @@
+import React, { useState, useEffect } from "react";
+function TodoItem({ todo, onDelete }) {
+  const [isChecked, setIsChecked] = useState(false);
+
+  const changeOnChecked = (event) => {
+    setIsChecked(event.target.checked);
+  };
+
+  useEffect(() => {
+    const spanElement = document.getElementById(`span${todo.id}`);
+    if (isChecked) {
+      spanElement.style.textDecoration = "line-through";
+    } else {
+      spanElement.style.textDecoration = "none";
+    }
+  }, [isChecked, todo.id]);
+
+  return (
+    <li>
+      <span id={`span${todo.id}`}>{todo.description}</span>
+      <input type="checkbox" checked={isChecked} onChange={changeOnChecked} />
```

Comment:
I think, instead of onChange, onClick makes more meaning as it is a checked type of input.